src/entities/queries.rs

```diff
@@ -41,3 +41,18 @@ WHERE
           )
       ) as friends_b
   );";
+
+/// This query fetches the rows where the last event of a friendship_id is a REQUEST, and either address_1 or address_2 is equal to the given user's address.
+pub const USER_REQUESTS_QUERY: &str = "SELECT f.address_1, f.address_2, 
+fh.acting_user, fh.timestamp, fh.metadata
+FROM friendship_history fh
+  JOIN (
+    SELECT friendship_id, MAX(timestamp) AS last_timestamp
```

Comment:
This is selecting the latest request that has been sent or received but it doesn't check whether it was accepted/rejected, right?

Maybe what you need here is:
1. Get all friendships_ids from the given user
2. Get the latest interaction of each friendship
3. Filter only interactions that are from the Request event

Reply:
I thought I was doing that haha. I'll check it 😛